packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2419,6 +2419,17 @@ class RenderTransform extends RenderProxyBox {
     this.origin = origin;
   }
 
+  @override
+  OffsetLayer updateCompositedLayer({required covariant ImageFilterLayer? oldLayer}) {
+    final ImageFilterLayer layer = oldLayer ?? ImageFilterLayer();
+    final Matrix4 transform = _effectiveTransform!;
+    layer.imageFilter = ui.ImageFilter.matrix(
```

Comment:
Where is the offset? The matrix for ImageFilter is incorrect unless the offset is either applied by the layer or accounted for in the computation of the transform it uses. The translate/-translate code that was eliminated in the paint method used to do this.

Reply:
The offset is applied directly to the ImageFilterLayer by the framework. Since it will be a repaint boundary if the filter quality is used that allows the framework to change the offset without repainting the children

Reply:
Where does that happen?

Reply:
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/rendering/object.dart#L118-L175

Reply:
actually that's not quite correct, that assumes we're starting at a root, one sec

Reply:
here: https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/rendering/object.dart#L257-L280

Reply:
I see now, that looks fine then - but the goldens are still a problem so my "needs changes" review still stands until the golden tests pass. (see comment below)